GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -667,7 +675,9 @@ public void beforeDrawEyes() {
 
             doMemoryManagementAndPerFrameCallbacks();
 
-            mScript.onStep();
+            GVRViewManager.this.getEventManager().sendEvent(
```

Comment:
Good to know the profiling result. We need to use this indirection for event calls. In typical event-based systems, the latency may event be longer. I doubt moving script off GL thread works, because threads need to manipulate scene / scene objects. Caching would be the right way to go.